lib/internal/modules/esm/resolve.js

```diff
@@ -933,6 +922,12 @@ function shouldBeTreatedAsRelativeOrAbsolutePath(specifier) {
   return isRelativeSpecifier(specifier);
 }
 
+function amendFormatToUrl(resolved) {
+  const format = getModuleFileFormat(resolved, false);
+
+  return { resolved, ...(format !== null) && { format } };
```

Comment:
Can you add this code snippet at the top of `test/es-module/test-esm-resolve-type.js` please?

```js
Reflect.defineProperty(Object.prototype, 'format', {
  get: common.mustNotCall('get Object.prototype.format'),
})
```

Reply:
I cannot commit this because the tests would fail. For example [here](https://github.com/dynatrace-oss-contrib/node/blob/a182a2163606c4b7433801a7b07c7db1fbf6b653/lib/internal/modules/esm/resolve.js#L1059).